Robovm compiler crashes with Kotlin modules when compiling for debug

#### Issue details
A Robovm project which includes Kotlin code crashes on compilation in debug mode.

#### Reproduction steps/code
In Android Studio or Idea create a new Robovm iOS project. Add a Kotlin class to the project. Create a run configuration and click the "Debug" button (or select Run/Debug ...)

#### Expected result
App will run on the simulator/device and be debuggable

#### Observed behaviour
AS/Idea crashes

#### Configuration

**Build Tools:**
Android Studio 3.0
 
**Versions:**

_Please provide the version of RoboVM, XCode and JDK used_
- Robovm: 2.3.4-SNAPSHOT
- XCode: 9.3
- JDK: 1.8

**Build Targets:**

Any iOS build target

---

#### Stacktrace
```java
Thread 87 Crashed:: Java: pool-20-thread-3
0   libsystem_kernel.dylib              0x00007fff78fbab6e __pthread_kill + 10
1   libsystem_pthread.dylib             0x00007fff79185080 pthread_kill + 333
2   libsystem_c.dylib                   0x00007fff78f161ae abort + 127
3   libjvm.dylib                        0x000000010748faa8 os::abort(bool) + 22
4   libjvm.dylib                        0x0000000107598b2d VMError::report_and_die() + 2207
5   libjvm.dylib                        0x0000000107494116 JVM_handle_bsd_signal + 511
6   libjvm.dylib                        0x00000001074919dc signalHandler(int, __siginfo*, void*) + 45
7   libsystem_platform.dylib            0x00007fff79178f5a _sigtramp + 26
8   ???                                 0x00006080036f4b00 0 + 106102929705728
9   librobovm-llvm1310633165281695321.dylib     0x000000018be9950b llvm::DwarfDebug::collectDeadVariables() + 427
10  librobovm-llvm1310633165281695321.dylib     0x000000018be995bc llvm::DwarfDebug::finalizeModuleInfo() + 124
11  librobovm-llvm1310633165281695321.dylib     0x000000018be99d10 llvm::DwarfDebug::endModule() + 48
12  librobovm-llvm1310633165281695321.dylib     0x000000018be7f187 llvm::AsmPrinter::doFinalization(llvm::Module&) + 1847
13  librobovm-llvm1310633165281695321.dylib     0x000000018c508ae5 llvm::FPPassManager::doFinalization(llvm::Module&) + 53
14  librobovm-llvm1310633165281695321.dylib     0x000000018c508fec llvm::legacy::PassManagerImpl::run(llvm::Module&) + 1228
15  librobovm-llvm1310633165281695321.dylib     0x000000018b947c05 LLVMTargetMachineEmitToOutputStream + 421
16  librobovm-llvm1310633165281695321.dylib     0x000000018b946817 Java_org_robovm_llvm_binding_LLVMJNI_TargetMachineEmitToOutputStream + 71
17  ???                                 0x0000000116999cb2 0 + 4674133170
18  ???                                 0x000000011b27ae74 0 + 4750552692
```

Wow that was quick. I tested the patch on my sample project, that fixes it. On my full project the compile now completes successfully, but I get another error at launch with the debugger:

````
[ERROR] Couldn't start application
java.lang.IllegalArgumentException: there is no region to read string @1030032500000
	at org.robovm.debugger.utils.bytebuffer.ByteBufferMemoryReader.readStringZAtPtr(ByteBufferMemoryReader.java:94)
	at org.robovm.debugger.state.classdata.ClassInfoImpl.readClassInfoHeader(ClassInfoImpl.java:119)
	at org.robovm.debugger.state.classdata.ClassInfoLoader.parseHash(ClassInfoLoader.java:98)
	at org.robovm.debugger.state.classdata.ClassInfoLoader.<init>(ClassInfoLoader.java:70)
	at org.robovm.debugger.state.VmDebuggerState.<init>(VmDebuggerState.java:73)
	at org.robovm.debugger.Debugger.<init>(Debugger.java:76)
	at org.robovm.debugger.DebuggerLaunchPlugin.afterLaunch(DebuggerLaunchPlugin.java:136)
	at org.robovm.compiler.AppCompiler.launchAsync(AppCompiler.java:837)
	at org.robovm.idea.running.RoboVmRunProfileState.executeRun(RoboVmRunProfileState.java:69)
	at org.robovm.idea.running.RoboVmRunProfileState.startProcess(RoboVmRunProfileState.java:128)
	at com.intellij.execution.configurations.CommandLineState.execute(CommandLineState.java:80)
	at com.intellij.debugger.DefaultDebugEnvironment.createExecutionResult(DefaultDebugEnvironment.java:92)
	at com.intellij.debugger.engine.DebugProcessImpl.attachVirtualMachine(DebugProcessImpl.java:1873)
	at com.intellij.debugger.impl.DebuggerSession.attach(DebuggerSession.java:447)
	at com.intellij.debugger.impl.DebuggerSession.create(DebuggerSession.java:208)
	at com.intellij.debugger.impl.DebuggerManagerImpl.attachVirtualMachine(DebuggerManagerImpl.java:221)
	at org.robovm.idea.running.RoboVmRunner.attachVirtualMachine(RoboVmRunner.java:93)
	at org.robovm.idea.running.RoboVmRunner.doExecute(RoboVmRunner.java:77)
	at com.intellij.execution.runners.GenericProgramRunner$1.execute(GenericProgramRunner.java:40)
	at com.intellij.execution.RunProfileStarter.executeAsync(RunProfileStarter.java:43)
	at com.intellij.execution.impl.ExecutionManagerImpl.lambda$startRunProfile$7(ExecutionManagerImpl.java:390)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:86)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$pollQueueLater$0(TransactionGuardImpl.java:72)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:410)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:399)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:827)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:655)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:365)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
````

Shall I open a new issue?

--- a/compiler/compiler/src/main/java/org/robovm/compiler/plugin/debug/DebugInformationPlugin.java
+++ b/compiler/compiler/src/main/java/org/robovm/compiler/plugin/debug/DebugInformationPlugin.java
@@ -265,9 +265,15 @@ public class DebugInformationPlugin extends AbstractCompilerPlugin {
             }
         }
 
+        classBundle.diMethods.add(diSubprogram);
+        if (methodLineNumber == Integer.MAX_VALUE) {
+            // there was no debug information for this method
+            // it will be not possible to resolve variables, just return
+            return;
+        }
+
         diSubprogram.setScopeLineNo(methodLineNumber);
         diSubprogram.setDefLineNo(methodLineNumber);
-        classBundle.diMethods.add(diSubprogram);
 
         if (!includeDebuggerInfo) {
             return;

--- a/plugins/debugger/src/main/java/org/robovm/debugger/state/classdata/ClassInfoImpl.java
+++ b/plugins/debugger/src/main/java/org/robovm/debugger/state/classdata/ClassInfoImpl.java
@@ -95,6 +95,18 @@ public class ClassInfoImpl extends ClassInfo {
         className = reader.readStringZAtPtr(ptr);
         signature = "L" + className + ";";
 
+        if (hasError()) {
+// Uncomment to get information what is broken
+//            int errorType = reader.readInt32();
+//            ptr = reader.readPointer(true);
+//            String errorMessage = null;
+//            if (ptr != 0) {
+//                errorMessage = reader.readStringZAtPtr(ptr);
+//            }
+//            System.out.println("!Class " + className + " has error " + errorType + " due " + errorMessage);
+            return;
+        }
+
         // read other fields
         reader.skip(pointerSize + //    void* initializer;
                 pointerSize + //    TypeInfo* typeInfo;
@@ -123,6 +135,10 @@ public class ClassInfoImpl extends ClassInfo {
     }
 
     void loadData(ClassInfoLoader loader) {
+        // skip if class is broken
+        if (hasError())
+            return;
+
         // set to zero if already read
         if (endOfHeaderPos == 0)
             return;

